### ISSUE TITLE
Uncaught TypeError: Cannot read property 'value' of null

### ISSUE THREAD

**Issue by wolfadex** [nikhilkalige/docblockr#197]
I attempted to add comments to an expressjs route.

1. Create route ```views.get('/', isLoggedIn, function (req, res) {...```
2. On line above route type ```/**``` and then press enter

**Atom Version**: 1.9.6
**System**: Mac OS X 10.11.4
**Thrown From**: [docblockr](https://github.com/nikhilkalige/docblockr) package, v0.7.4

### Stack Trace

Uncaught TypeError: Cannot read property 'value' of null

```
At /Users/wolfgang.schuster/.atom/packages/docblockr/lib/languages/javascript.js:81

TypeError: Cannot read property 'value' of null
    at DocsParser.JsParser.get_arg_type (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/languages/javascript.js:81:15)
    at DocsParser.parse_args (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/docsparser.js:265:24)
    at DocsParser.format_function (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/docsparser.js:130:32)
    at DocsParser.parse (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/docsparser.js:44:37)
    at DocBlockrAtom.module.exports.DocBlockrAtom.parse_command (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/docblockr-worker.js:287:29)
    at atom-workspace.<anonymous> (/Users/wolfgang.schuster/.atom/packages/docblockr/lib/docblockr-worker.js:77:16)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:00.2.0 docblockr:parse-enter (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "disabledPackages": [
      "autosave",
      "atom-jshint",
      "jshint"
    ],
    "excludeVcsIgnoredPaths": false,
    "themes": [
      "genesis-ui",
      "genesis-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
.bin, vundefined (inactive)
atom-html-preview, v0.1.22 (active)
atom-jshint, v2.0.0 (inactive)
autoprefixer, v3.2.0 (inactive)
base16-syntax, v1.7.0 (inactive)
color-picker, v2.2.2 (active)
docblockr, v0.7.4 (active)
file-icons, v1.7.18 (active)
genesis-syntax, v1.0.7 (active)
genesis-ui, v0.5.0 (active)
git-history, v3.2.0 (inactive)
git-plus, v5.16.2 (inactive)
gutter-shadow, v0.6.0 (active)
highlight-selected, v0.11.2 (active)
htmlhint, v1.1.4 (active)
jshint, v1.8.5 (inactive)
language-babel, v2.32.2 (active)
language-lua, v0.9.4 (active)
language-pug, v0.0.17 (active)
less-autocompile, v1.0.0 (active)
less-than-slash, v0.16.0 (active)
linter, v1.11.15 (active)
linter-eslint, v7.2.4 (active)
minimap, v4.24.7 (active)
minimap-pigments, v0.2.1 (active)
monokai-seti, v0.7.0 (inactive)
pigments, v0.31.2 (active)
react-snippets, v0.4.2 (active)
Remote-FTP, v0.8.3 (active)
sass-autocompile, v0.12.8 (active)
set-syntax, v0.3.0 (active)
seti-monokai, v0.3.1 (inactive)
seti-ui, v1.3.2 (inactive)
simple-align, v0.2.0 (inactive)
sort-lines, v0.14.0 (inactive)
spacegray-eighties-ui, v1.2.0 (inactive)
Sublime-Style-Column-Selection, v1.7.2 (active)
svg-preview, v0.8.0 (active)
visual-editor, v0.0.1 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (inactive)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (inactive)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.0 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (inactive)
background-tips, v0.26.0 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.38.3 (active)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.64.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.0 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.46.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.0 (active)
timecop, v0.33.1 (active)
tree-view, v0.208.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.32.2 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.36.2 (active)
language-gfm, v0.86.0 (active)
language-git, v0.13.0 (active)
language-go, v0.42.0 (active)
language-html, v0.44.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.22.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.0 (active)
language-less, v0.29.3 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.0 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.68.5 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.52.0 (active)
language-shellscript, v0.22.3 (active)
language-source, v0.9.0 (active)
language-sql, v0.21.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.8 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by wolfadex** [comment]
This could be related to issues #195 and #196.

**Comment by MoritzKn** [comment]
I really thought #199 would fix this -- it didn't.
The problem occurs because the function name `get` also is a keyword for declaring functions.
[5bf99840](https://github.com/nikhilkalige/docblockr/commit/5bf998402831875ec99da3def2ae244c0be05fda) should finally fix this.

**Comment by sompylasar** [comment]
Still getting it. The cause is the trailing comma in the destructuring object pattern.

Minimal repro:
```
/**
export function foo({x,}) {}
```
Cursor after `/**`, press ENTER.

**Atom Version**: 1.9.9
**System**: Mac OS X 10.11.4
**Thrown From**: [docblockr](https://github.com/nikhilkalige/docblockr) package, v0.8.4


### Stack Trace

Uncaught TypeError: Cannot read property 'value' of null

```
At /Users/sompylasar/.atom/packages/docblockr/lib/languages/javascript.js:128

TypeError: Cannot read property 'value' of null
    at DocsParser.JsParser.get_arg_name (/Users/sompylasar/.atom/packages/docblockr/lib/languages/javascript.js:128:16)
    at DocsParser.parse_args (/Users/sompylasar/.atom/packages/docblockr/lib/docsparser.js:265:48)
    at DocsParser.format_function (/Users/sompylasar/.atom/packages/docblockr/lib/docsparser.js:130:32)
    at DocsParser.parse (/Users/sompylasar/.atom/packages/docblockr/lib/docsparser.js:44:37)
    at DocBlockrAtom.module.exports.DocBlockrAtom.parse_command (/Users/sompylasar/.atom/packages/docblockr/lib/docblockr-worker.js:287:29)
    at atom-workspace.<anonymous> (/Users/sompylasar/.atom/packages/docblockr/lib/docblockr-worker.js:77:16)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  2x -0:03.8.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -0:03.3.0 core:move-left (atom-text-editor.editor.is-focused)
  3x -0:02.9.0 core:select-down (atom-text-editor.editor.is-focused)
  3x -0:02.5.0 core:select-right (atom-text-editor.editor.is-focused)
     -0:01.5.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:00.1.0 docblockr:parse-enter (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "allowPendingPaneItems": false,
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.js.jsx": [
        ".js"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "metrics",
      "atom-fix-path",
      "terminal-plus",
      "color-picker",
      "git-control",
      "atom-beautify",
      "atom-runner",
      "git-blame",
      "git-clone",
      "git-plus",
      "git-time-machine",
      "autocomplete-modules",
      "bookmarks",
      "linter-jshint",
      "nuclide",
      "linter-htmlhint",
      "remote-sync",
      "autocomplete-snippets"
    ],
    "projectHome": "/Users/sompylasar/_github",
    "themes": [
      "atom-material-ui",
      "one-dark-syntax"
    ]
  },
  "docblockr": {
    "extend_double_slash": false,
    "lower_case_primitives": true,
    "spacer_between_sections": true
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.10 (inactive)
atom-cursor-indent, v0.1.1 (active)
atom-fix-path, v0.1.0 (inactive)
atom-material-syntax, v0.4.6 (inactive)
atom-material-ui, v1.3.3 (active)
atom-runner, v2.6.0 (inactive)
atom-ternjs, v0.14.2 (active)
autocomplete-modules, v1.6.1 (inactive)
blackspace, v1.1.0 (active)
color-picker, v2.2.2 (inactive)
docblockr, v0.8.4 (active)
editorconfig, v1.4.1 (active)
file-icons, v1.7.18 (active)
filesize, v0.4.2 (active)
git-blame, v0.4.11 (inactive)
git-clone, v0.1.5 (inactive)
git-control, v0.8.2 (inactive)
git-plus, v5.16.2 (inactive)
git-time-machine, v1.5.3 (inactive)
highlight-bad-chars, v1.0.5 (active)
highlight-line, v0.11.1 (active)
highlight-selected, v0.11.2 (active)
hyperclick, v0.0.37 (active)
indent-guide-improved, v1.4.9 (active)
js-hyperclick, v1.4.2 (active)
language-babel, v2.35.0 (active)
linter, v1.11.15 (active)
linter-csslint, v1.3.2 (active)
linter-eslint, v7.2.4 (active)
linter-htmlhint, v1.3.0 (inactive)
linter-jshint, v2.1.0 (inactive)
linter-mixed-indent, v1.1.0 (active)
linter-sass-lint, v1.7.0 (active)
lucid-tabs, v1.1.1 (active)
merge-conflicts, v1.4.4 (active)
minimap, v4.24.7 (active)
minimap-cursorline, v0.2.0 (active)
minimap-find-and-replace, v4.5.1 (active)
minimap-highlight-selected, v4.4.0 (active)
minimap-linter, v1.1.1 (active)
minimap-pigments, v0.2.1 (active)
nice-index, v1.0.2 (active)
nuclide, v0.159.0 (inactive)
open-terminal-here, v2.1.4 (inactive)
pane-layout-plus, v0.7.3 (inactive)
pigments, v0.31.2 (active)
pinned-tabs, v0.3.1 (active)
remote-sync, v4.1.2 (inactive)
rest-client, v1.2.1 (active)
set-syntax, v0.3.0 (active)
simple-drag-drop-text, v0.3.1 (active)
svg-preview, v0.8.0 (active)
sync-settings, v0.7.2 (active)
tabulation-control, v1.0.7 (active)
terminal-plus, v0.14.5 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (inactive)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (active)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.0 (active)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.0 (active)
bookmarks, v0.41.0 (inactive)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.1 (active)
metrics, v0.53.1 (inactive)
notifications, v0.64.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.1 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.1 (active)
timecop, v0.33.1 (active)
tree-view, v0.208.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.32.2 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.36.2 (active)
language-gfm, v0.86.0 (active)
language-git, v0.13.0 (active)
language-go, v0.42.0 (active)
language-html, v0.44.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.22.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.0 (active)
language-less, v0.29.3 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.0 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.68.5 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.52.0 (active)
language-shellscript, v0.22.3 (active)
language-source, v0.9.0 (active)
language-sql, v0.21.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.8 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```


By the way (that's another issue), the destructuring pattern is not fully supported, I get `}` at the end of the default value spec:
```
/**
 * [foo description]
 *
 * @param  {[type]} [x=false}] [description]
 *
 * @return {[type]}            [description]
 */
export function foo({x=false}) {}
```

**Comment by MoritzKn** [comment]
@sompylasar The root of this problem is that destructuring is not yet supported. See #180.
